File: frontend/src/services/xxjs/dianming.ts
import { api } from '../api';

export interface DianmingClass {
  year: string;
  class_name: string;
  count: number;
}

export interface DianmingStudent {
  id: number;
  year: string;
  class_name: string;
  student_name: string;
  student_no?: string;
  created_at: string;
}

export interface DianmingImportRequest {
  year: string;
  class_name: string;
  names_text: string;
}

const API_PREFIX = '/xxjs/dianming';

export const dianmingApi = {
  // 获取班级列表
  listClasses: async () => {
    const response = await api.client.get<DianmingClass[]>(`${API_PREFIX}/classes`);
    return response.data;
  },

  // 获取指定班级学生
  listStudents: async (year: string, class_name: string) => {
    const response = await api.client.get<DianmingStudent[]>(`${API_PREFIX}/students`, {
      params: { year, class_name },
    });
    return response.data;
  },

  // 批量导入学生
  importStudents: async (data: DianmingImportRequest) => {
    const response = await api.client.post<DianmingStudent[]>(`${API_PREFIX}/import`, data);
    return response.data;
  },
  
  // 更新班级学生名单 (覆盖)
  updateClassStudents: async (data: DianmingImportRequest) => {
    const response = await api.client.put<DianmingStudent[]>(`${API_PREFIX}/class/students`, data);
    return response.data;
  },

  // 删除班级
  deleteClass: async (year: string, class_name: string) => {
    const response = await api.client.delete<{ success: boolean }>(`${API_PREFIX}/class`, {
      params: { year, class_name },
    });
    return response.data;
  },
};

File: backend/app/schemas/xxjs/dianming.py
"""
信息技术 - 点名系统 Pydantic Schemas
"""

from typing import Any, Optional

from pydantic import BaseModel, ConfigDict


class DianmingClass(BaseModel):
    """班级聚合视图"""
    year: str
    class_name: str
    count: int


class DianmingStudentBase(BaseModel):
    """学生基础字段"""
    year: str
    class_name: str
    student_name: str
    student_no: Optional[str] = None


class DianmingStudentCreate(DianmingStudentBase):
    """创建学生（预留扩展）"""
    pass


class DianmingStudent(DianmingStudentBase):
    """学生完整响应"""
    id: int
    created_at: Any
    model_config = ConfigDict(from_attributes=True)


class DianmingImportRequest(BaseModel):
    """批量导入请求"""
    year: str
    class_name: str
    names_text: str  # 纯文本，换行分隔
